examples/basic_usage.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(stmt_expr_attributes, proc_macro_hygiene)]
+
+use safe::safe;
+
+fn main() {
+    let mut x: u32 = 42;
+    let x_ptr = &mut x as *mut u32;
+
+    #[safe(reason = "This is the only reference to x")]
+    unsafe {
+        x_ptr.write(7);
+    }
+
+    #[safe(
+        reason = "This is the only reference to x",
+        precondition = "!x_ptr.is_null()"
```

Comment:
I believe [Dafny uses `requires`](https://rise4fun.com/Dafny/tutorialcontent/guide#h22) and [JML does as well](https://en.wikipedia.org/wiki/Java_Modeling_Language#Syntax) -- that might be better?

Reply:
I was originally thinking of having `precondition` and `postcondition` because you'll often have invariants which must be held before and after the `unsafe` block. If it were called `requires` then we'd miss out on that nice symmetry.

Reply:
So in Dafny and JML I believe postconditions are called `ensures` so that fits?

Reply:
Fair point :+1: 

I've never used Dafny or JML, so didn't know there was existing terminology for it.